h5py_wrapper/wrapper.py

```diff
@@ -259,6 +259,8 @@ def _load_dataset(f, lazy=False):
     else:
         try:
             value_type = f.attrs['_value_type']
+            if isinstance(value_type, bytes):
+                value_type = str(value_type, 'utf-8')
```

Comment:
We need to provide 'utf-8' because otherwise python3 puts the 'b' of the bytes string into the string.
Try:
`str(b'test')` will lead to `"b'test'"`

Yes, it fails when applied to a string. That's why I introduced the type check before, testing whether `isinstance(value_type, bytes)`.